packages/TelegramClient-UI.package/TCUEmojiHelper.class/class/areEmojisPresent.st

```diff
@@ -0,0 +1,4 @@
+downloading
+areEmojisPresent
+
+	^ self defaultDirectory fileExists: self defaultCanaryName
```

Comment:
Instead of the canary, could you maybe create a string hash over all emoji file names? Something like joining all filenames and getting a hash value over that string and comparing it to a default value. Because if I now manually or by accident remove screenshots but keep the canary.txt, nothing will be redownloaded.

Reply:
We have to update the hash every time the list of emojis changes. Other ideas:
1) check for a particular emoji and assume the other emojis are also present
2) concatenate all emoji names to a string and calculate levenstein distance
3) check if the number of emoji files in the emojis folder is greater than some number eg. 1000

Reply:
> by accident remove screenshots but keep the canary.txt, nothing will be redownloaded.
if you manage to do that, its your own fault 😋

am implementing tom's third solution now, as it seems the easiest and most readable

Reply:
Fair enough, albeit a bit of a workaround 😛